big_tests/tests/inbox_SUITE.erl

```diff
@@ -678,6 +681,55 @@ simple_groupchat_stored_in_all_inbox(Config) ->
         check_inbox(Kate, [#conv{unread = 1, from = AliceRoomJid, to = KateJid, content = Msg}])
       end).
 
+remove_muclight_messages_after_the_room_is_deleted_xmpp(Config) ->
+    remove_muclight_messages_after_the_room_is_deleted(Config, xmpp).
+
+remove_muclight_messages_after_the_room_is_deleted_rest(Config) ->
+    remove_muclight_messages_after_the_room_is_deleted(Config, rest).
+
+remove_muclight_messages_after_the_room_is_deleted(Config, ReqType) ->
+    escalus:fresh_story(Config, [{alice, 1}, {bob, 1}, {kate, 1}], fun(Alice, Bob, Kate) ->
+        Msg = <<"Hi Room!">>,
+        Id = <<"MyID">>,
+        Users = [Alice, Bob, Kate],
+        Room = inbox_helper:create_room(Alice, [Bob, Kate]),
+        AliceJid = inbox_helper:to_bare_lower(Alice),
+        KateJid = inbox_helper:to_bare_lower(Kate),
+        BobJid = inbox_helper:to_bare_lower(Bob),
+        RoomJid = room_bin_jid(Room),
+        AliceRoomJid = <<RoomJid/binary, "/", AliceJid/binary>>,
+        Stanza = escalus_stanza:set_id(
+                   escalus_stanza:groupchat_to(RoomJid, Msg), Id),
+        %% Alice sends message to a room
+        escalus:send(Alice, Stanza),
+        inbox_helper:wait_for_groupchat_msg(Users),
+        %% Alice has 0 unread messages
+        check_inbox(Alice, [#conv{unread = 0, from = AliceRoomJid, to = AliceJid, content = Msg}]),
+        %% Bob and Kate have one conv with 1 unread message
+        check_inbox(Bob, [#conv{unread = 1, from = AliceRoomJid, to = BobJid, content = Msg}]),
+        check_inbox(Kate, [#conv{unread = 1, from = AliceRoomJid, to = KateJid, content = Msg}]),
+
+        %% Remove the room
+        destroy_room(Alice, Bob, Kate, Room, ReqType),
+
+        %% Inbox entries are removed
+        check_inbox(Alice, []),
+        check_inbox(Bob, []),
+        check_inbox(Kate, [])
+      end).
+
+destroy_room(Alice, Bob, Kate, Room, xmpp) ->
+    Stanza = escalus_stanza:to(
+                escalus_stanza:iq_set(?NS_MUC_LIGHT_DESTROY, []),
+                room_bin_jid(Room)),
+    escalus:send(Alice, Stanza),
+    AffUsersChanges = [{Alice, none}, {Bob, none}, {Kate, none}],
+    muc_light_helper:verify_aff_bcast([], AffUsersChanges, [?NS_MUC_LIGHT_DESTROY]);
+destroy_room(_, _, _, Room, rest) ->
+    Host = muc_light_helper:muc_host(),
+    Path = <<"/muc-lights/", Host/binary, "/", Room/binary, "/management">>,
+    {{<<"204">>, <<"No Content">>}, <<"">>} = rest_helper:delete(admin, Path).
+
```

Comment:
Maybe we could use GraphQL as we are going to phase out REST soon? This would be one less thing to convert in the future.